### ISSUE TITLE
ObjectUnsubscribedError when effects yield finite stream

### ISSUE THREAD

**Issue by samoshkin** [ngrx/effects#30]
Maybe, it's a bit uncommon case in real-world project, but can knock you down, when you're setting up and wiring all things together at the beginning of the project. 

Suppose, we have `Effect` class, which declares finite stream of single action (e.g, emit actions on init to  kickstart all things)

```
@Injectable()
export class TaskEffects {

  @Effect()
  loadTasksOnInit$ = Observable.of(loadAllTasks());
}
```

Then, somewhere else, you might have an effect or just regular Observable, which you'd like to manually wire to the Store.

```
constructor(private store: Store<AppState>, private effect: AnotherTaskEffects) {
    this.effect.fake$.subscribe(this.store);

    // or maybe just wire regular Observable to Store
    Observable.of({ type: 'action/fake' }).delay(1000).subscribe(this.store)
}
```

As a result, you would get `ObjectUnsubscribedError` with a huge stack trace, with no single application code frame. That is extremely frustrating! 

So, what happens, is that all streams from Effect classes are eventually merged and are subscribed to the store. 

```
mergeEffects(...effects).subscribe(store);
```

And when no effect streams or only finite effect streams are declared, merge operator produce finite stream as well, which eventually yields `store.complete()` when subscribed. At this point, Store is completed, and further `store.next()` or `obs.subscribe(store)` are invalid. Meet `ObjectUnsubscribedError`.

It's pretty easy to fix, but throwing in yet another effect with infinite stream.

```
@Effect()
infinite = Observable.never();
```

But I'm sure, that it should be `ngrx/effecs` job to ensure that stream is always infinite and does not break Store. Not the application developer concern. 

```
export function connectEffectsToStore(store: Store<any>, effects: any[]) {
  return function() {
    mergeEffects(...effects, Observable.never()).subscribe(store);

    return Promise.resolve(true);
  };
}
```

**Comment by awerlang** [comment]
Dupe of #24

**Comment by awerlang** [comment]
Beginning rxjs 5.0.0-beta.8, due to this [commit](https://github.com/ReactiveX/RxJS/commit/ba9ef2b),

> BREAKING CHANGE: Subjects no longer automatically unsubscribe when completed or errored

I wish ngrx/store maintains this property. Btw, upgrading to beta.10 (latest) and renaming a couple methods (_next => next, _error => error on SyncSubject), both #24 and #30 are resolved then.

**Comment by Hyperonic** [comment]
@awerlang 

can you please tell which changes did u make in SynSubject of @ngrx/core ? I tried renaming some methods but it didn't work.

I'm getting this 

@ngrx/core/SyncSubject.d.ts(4,22): Error TS2415: Class 'SyncSubject<T>' incorrectly extends base class 'Subject<T>'.
  Types of property '_subscribe' are incompatible.
    Type '(subscriber: Subscriber<T>) => AnonymousSubscription | Function | void' is not assignable to type '(subscriber: Subscriber<T>) => Subscription'.
      Type 'AnonymousSubscription | Function | void' is not assignable to type 'Subscription'.
        Type 'AnonymousSubscription' is not assignable to type 'Subscription'.
          Property 'closed' is missing in type 'AnonymousSubscription'.

**Comment by awerlang** [comment]
I suggest you update to latest beta.3 release of effects.

**Comment by MikeRyan52** [comment]
I think this should be fixed in v2.0.0 with the latest rxjs beta. Let me know if you still encounter problems.
